### ISSUE TITLE
preventCurveOverShooting sounds like mitigation rather than solution

### ISSUE THREAD

**Issue by chris-rutkowski** [imaNNeo/fl_chart#1648]
Hey, amazing plugin, don't get me wrong with this issue.

There are a lot of [issues](https://github.com/imaNNeo/fl_chart/issues?q=is%3Aissue+preventCurveOverShooting+) that include the keyword `preventCurveOverShooting`. It is always suggested to the problem that can be seen on the picture (take from one of the issues):

<img width="152" alt="Screenshot 2024-05-01 at 9 06 43 PM" src="https://github.com/imaNNeo/fl_chart/assets/58718695/55cc259a-5e47-4047-b49e-bf3dafd40af2">

But I don't think it is a solution. While it's nice to have a control over "roundness", there shouldn't be `preventCurveOverShooting+preventCurveOvershootingThreshold` as there shouldn't be any over shooting in the first place. Chart shouldn't show data that doesn't exist.

Those bumps are buggy and glitchy. `aboveBarData` and `belowBarData` doesn't colour them they are beyond bounds. The issue https://github.com/imaNNeo/fl_chart/issues/25 has been closed, despite glitch issued there by @jamesblasco which clearly indicates line getting thinner in some part when those `prevent...` features are toggled on.

Once again, amazing work, love your plugin, but would kindly appreciate reconsidering this implementation. We just want round lines, not zigzags :)

**Comment by TobiasRump** [comment]
Yeah it would be a really good idea, to have the `preventCurveOverShooting = true` as default. 

Here is again a user from today, who is confused

https://github.com/imaNNeo/fl_chart/issues/1649

**Comment by cedvdb** [comment]
@imaNNeo 

Curves are fine. The issue is problematic when the curve goes below a bottom or an upper threshold. 

For instance if you have 0% and 100%, it does not make sense to have the line go beyond those two thresholds.

Same if you are displaying sales per hour:

![image](https://github.com/user-attachments/assets/be5703fa-8910-45bb-8393-48a4acc63b75)


Imo a better solution would be to have bounds for the data so it does not go beyond those bounds.

What do you think ?
